### ISSUE TITLE
文本分类训练出现错误

### ISSUE THREAD

**Issue by wanggei** [PaddlePaddle/PaddleNLP#4630]
### 请提出你的问题

进行文本训练时，训练一段时间，就会出现这样的问题

paddlepaddle：2.3.2

paddlenlp：2.5.0

执行的train.py也是用最新的

![c5717dd465446bed693e681a008fff0](https://user-images.githubusercontent.com/61741330/216363239-9b21d028-b3c4-4681-ae81-43ead17748f4.png)

**Comment by lugimzzz** [comment]
我在本地测试并没有发现类似问题。请问你执行文件链接和执行命令是什么？看上去是设置的compute_metrics中不存在eval_accuracy或accuracy

**Comment by wanggei** [comment]
@lugimzzz 执行命令为python train.py --model_name_or_path ernie-3.0-base-zh --data_dir data --output_dir checkpoint --device cpu --learning_rate 3e-5 --early_stopping_patience 4 --max_seq_length 512 --per_device_eval_batch_size 4 --per_device_train_batch_size 4 --num_train_epochs 40 --do_train True --do_eval True --metric_for_best_model accuracy --load_best_model_at_end True --evaluation_strategy epoch --save_strategy epoch --save_total_limit 1

执行文件直接定位到了E:\PaddleNLP-2.5.0\applications\text_classification\multi_class下再去执行上面的命令

**Comment by lugimzzz** [comment]
你首先检查一下PaddleNLP版本，确保安装的PaddleNLP版本是2.5.0
```
pip show paddlenlp
```
然后下载develop版本的PaddleNLP代码执行，具体操作:
```
git clone https://github.com/PaddlePaddle/PaddleNLP.git
cd PaddleNLP/applications/text_classification/multi_class
python train.py --model_name_or_path ernie-3.0-nano-zh --data_dir data --output_dir checkpoint --device cpu --learning_rate 3e-5 --early_stopping_patience 4 --max_seq_length 512 --per_device_eval_batch_size 4 --per_device_train_batch_size 4 --num_train_epochs 40 --do_train True --do_eval True --metric_for_best_model accuracy --load_best_model_at_end True --evaluation_strategy epoch --save_strategy epoch --save_total_limit 1
```
你看是否能正常使用，可以把训练集开发集改小，使用小模型先测试一下。这边我在paddlepaddle 2.3.2，paddlenlp2.5.0暂时没法复现出该问题。或可以提供一下aistudio项目等可复现该错误的链接，方便定位问题。

**Comment by wanggei** [comment]
行，我试试

**Comment by wanggei** [comment]
paddlenlp的版本是2,5,0

拉了最新的版本去运行，还是出现这个错误，我还换了一个电脑去测试，出现同样的错误

![image](https://user-images.githubusercontent.com/61741330/216522477-97518355-38f3-4273-b9ea-8e7ef3b1f18d.png)
![image](https://user-images.githubusercontent.com/61741330/216522533-3acea8a7-4332-40c0-af79-b9bc5fec5dc2.png)

**Comment by lugimzzz** [comment]
这块你到源码上打印一下metrics
![image](https://user-images.githubusercontent.com/63761690/216529789-e5951a21-111e-4e3f-8323-f658a3a303e7.png)

**Comment by wanggei** [comment]
尝试到哪里

**Comment by lugimzzz** [comment]
你是不是没有设置开发集dev.txt为空呀

**Comment by wanggei** [comment]
我开发集里面是没有数据，但是之前我在2.4.8版本是可以正常训练的，dev.txt里面一定要有数据吗？

**Comment by lugimzzz** [comment]
需要，因为这里参数需要根据开发集上的表现选择最佳模型

**Comment by wanggei** [comment]
好的，谢谢
